tests/integration/pipeline/test_bootstrap_command.py

```diff
@@ -0,0 +1,227 @@
+from unittest import skipIf
+
+from parameterized import parameterized
+
+from tests.integration.pipeline.base import BootstrapIntegBase
+from tests.testing_utils import (
+    run_command_with_input,
+    RUNNING_ON_CI,
+    RUNNING_TEST_FOR_MASTER_ON_CI,
+    RUN_BY_CANARY,
+    run_command,
+)
+
+# bootstrap tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_DEPLOY_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+
+
+@skipIf(SKIP_DEPLOY_TESTS, "Skip bootstrap tests in CI/CD only")
+class TestBootstrap(BootstrapIntegBase):
+    @parameterized.expand([("create_ecr_repo",), (False,)])
+    def test_interactive_with_no_resources_provided(self, create_ecr_repo: bool):
+        stage_name, stack_name = self._get_stage_and_stack_name()
+        self.stack_names = [stack_name]
+
+        bootstrap_command_list = self.get_bootstrap_command_list()
+
+        inputs = [
+            stage_name,
+            "",  # pipeline user
+            "",  # Pipeline execution role
+            "",  # CloudFormation execution role
+            "",  # Artifacts bucket
+            "2" if create_ecr_repo else "1",  # Should we create ECR repo, 1 - No, 2 - Yes
+            "",  # Pipeline IP address range
+            "y",  # proceed
+        ]
+
+        bootstrap_process_execute = self.run_command_with_inputs(bootstrap_command_list, inputs)
+
+        self.assertEqual(bootstrap_process_execute.process.returncode, 0)
+        stdout = bootstrap_process_execute.stdout.decode()
+        self.assertIn("We have created the following resources", stdout)
+        # make sure pipeline user's credential is printed
+        self.assertIn("ACCESS_KEY_ID", stdout)
+        self.assertIn("SECRET_ACCESS_KEY", stdout)
+
+        common_resources = {
+            "PipelineUser",
+            "PipelineUserAccessKey",
+            "CloudFormationExecutionRole",
+            "PipelineExecutionRole",
+            "ArtifactsBucket",
+            "ArtifactsBucketPolicy",
+            "PipelineExecutionRolePermissionPolicy",
+        }
+        if create_ecr_repo:
+            self.assertIn("arn:aws:ecr:", stdout)
+            self.assertSetEqual(
+                {
+                    *common_resources,
+                    "ECRRepo",
+                },
+                self._extract_created_resource_logical_ids(stack_name),
+            )
+        else:
+            self.assertNotIn("arn:aws:ecr:", stdout)
+            self.assertSetEqual(common_resources, self._extract_created_resource_logical_ids(stack_name))
+
+    @parameterized.expand([("create_ecr_repo",), (False,)])
```

Comment:
I'm quite confused about the "create_ecr_repo" string here, is this a way to pass a True value for the parameter (a non empty string is evaluated to True)? why not just passing True?

Reply:
It is our parameterized convention. By giving a string (which will resolved to `True` eventually` instead of a boolean will make the test name generated by pytest more readable (so that `create_ecr_repo` shows up in test name).

See how we did it in the past:

https://github.com/aws/aws-sam-cli/blob/5cc7680068c820e972d6165a0cccd21677e2a428/tests/integration/buildcmd/test_build_cmd.py#L90-L103